exercises/dominoes/src/test/java/DominoesTest.java

```diff
@@ -0,0 +1,201 @@
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		expectedException.expect(IllegalArgumentException.class);
+		expectedException.expectMessage("No domino chain found.");
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void disconnectedSingleIsolatedTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 3), new Domino(3, 1), new Domino(4, 4)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		expectedException.expect(IllegalArgumentException.class);
+		expectedException.expectMessage("No domino chain found.");
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void needBacktrackTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 3), new Domino(3, 1), new Domino(2, 4), new Domino(4, 2)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertValidChain(dominoesArray, chain);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void separateLoopsTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 3), new Domino(3, 1), new Domino(1, 1), new Domino(2, 2), new Domino(3, 3)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertValidChain(dominoesArray, chain);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void nineElementsTest() {
+		Dominoes dominoes = new Dominoes();
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(5, 3), new Domino(3, 1), new Domino(1, 2), new Domino(2, 4), new Domino(1, 6), new Domino(2, 3), new Domino(3, 4), new Domino(5, 6)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertValidChain(dominoesArray, chain);
+	}
+	
+	private void assertValidChain(ArrayList<Domino> inputDominoes, ArrayList<Domino> outputDominoes) {
+		assertSameDominoes(inputDominoes, outputDominoes);
+		assertEndDominoesMatch(outputDominoes);
+		assertConsecutiveDominoes(outputDominoes);
+	}
+	
+	private void assertEndDominoesMatch(ArrayList<Domino> outputDominoes) {
+		assertEquals("The end dominoes are not the same.", outputDominoes.get(0).getLeft(), outputDominoes.get(outputDominoes.size() - 1).getRight());
+	}
+	
+	private void assertSameDominoes(ArrayList<Domino> inputDominoes, ArrayList<Domino> outputDominoes) {
+		assertEquals("The number of dominoes in the output chain doesn't match the number in the input.", inputDominoes.size(), outputDominoes.size());
```

Comment:
Similarly to the assert above, and the same goes for the ones below, it would be great if you could include exactly which values are wrong in your error message 🙂